### ISSUE TITLE
Whether the performance of DiT improves as the VAE dimension increases?

### ISSUE THREAD

**Issue by LinB203** [hustvl/LightningDiT#3]
Please feel free to correct me if I have any misunderstandings. 

1. Although Figure 4 (a) and (b) demonstrate that +VF DINO V2 performs better, dim64 is not superior to dim32, am I right?
2. For the main results in Table 3, they should all use dim32 VAE, correct? I'm curious whether the performance of DiT improves as the VAE dimension increases. If not, then perhaps +VF DINO V2 simply shifts the trade-off from dim16 to dim32, which means it hasn't fundamentally addressed the issue.

![image](https://github.com/user-attachments/assets/9546dfcd-a3d6-4dcc-905d-1dedf2b6d557)

**Comment by yinghuozijin** [comment]
确实

**Comment by JingfengYao** [comment]
Thanks for your questions. 
1. Although low-dimensional tokenizers have a clear computational advantage in FID convergence, they inherently produce artifacts in detail reconstruction, such as small faces and tiny text (Figure 1 in the paper). Influential text-to-image models like FLUX and SD3 still opt for higher-dimensional tokenizers and larger generative models (>8B) to achieve finer generation quality.
2. In Table 2, VF loss significantly improves the convergence speed of high-dimensional tokenizers, whether d32 or d64. Although VF loss does not achieve performance improvements with increasing dimensions in small-scale models (<0.6B), its performance boost for specific high-dimensional tokenizers is still expected to reduce the parameter requirements for generative models. This effectively controls the optimization dilemma and advances the reconstruction-generation frontier.
3. Thank you again for your attention and feedback on our work. We will continue to refine our approach and ensure that our expressions in the paper are more precise.

<img width="524" alt="1736150662893" src="https://github.com/user-attachments/assets/e737f579-e761-4f97-aa73-7b00afeb27db" />

**Comment by LinB203** [comment]
This work is a meaningful exploration. I have a suggestion, feel free to critique. When the compression ratio remains unchanged, increasing the latent dimension from 16 to 64 might lead the VAE to exploit shortcuts instead of performing its intended task. This is because the reconstruction task becomes too easy in such cases. Perhaps methods should be devised to verify whether the VAE is truly learning meaningful representations.

**Comment by JingfengYao** [comment]
Great point, we'll consider it. Thanks~😄

**Comment by txytju** [comment]
<img width="804" alt="截屏2025-01-10 09 53 46" src="https://github.com/user-attachments/assets/36667b39-4ba0-4036-ac8a-f67224d7b9be" />
I tested a VAE that I trained with 64 channels, and found that the correlation coefficient between channels is very high.

**Comment by rakkit** [comment]
it is expected that f16d32 and f16d64 have different scale-factor, which could affect the Diffusion/FM training dynamic.
Does the code [here](https://github.com/hustvl/LightningDiT/blob/main/configs/lightningdit_xl_vavae_f16d32.yaml#L22) mean you use 1.0 for your DIT + VA-VAE training?

**Comment by JingfengYao** [comment]
@rakkit 
No, please refer [here](https://github.com/hustvl/LightningDiT/blob/main/inference.py#L220).

**Comment by JingfengYao** [comment]
@rakkit No, please refer to [img_latent_dataset](https://github.com/hustvl/LightningDiT/blob/main/datasets/img_latent_dataset.py#L83) and [config_details](https://github.com/hustvl/LightningDiT/blob/main/configs/config_details.yaml#L12) for details.

**Comment by rakkit** [comment]
Hi @JingfengYao, thanks for pointing this out.

I am surprised that vanilla-VAE works pretty well (rfid=0.49) with 50 epochs and no extra tricks. Though, it actually makes sense that by setting KL's weights to a very small amount number, we then have a degenerated VAE as plain AutoEncoder, which is supposed to max. performance on reconstruction.  This can also explain why we can refactor latent landscape via VF loss to improve generation performance. (Can you share the kl-weights seetings and stats. of your latents' mean and varience?)

also, it is not surprised that @LinB203  point out DiTB + D64 have worse performance than  DiTB+D32..  [SD3](https://arxiv.org/pdf/2403.03206) also have a similar conclusion  (figure 10, page 21).  But it is interesting that from SD3's conclusion, model >22 layers should benefit from high dimension latent, which is easy to stratify (DiT-L have 24 layers).  

The fact is we have already seen successful works in generations, and understanding tasks benefit from high-dimension,  and your methods should work as well.  The real problem could be that KL-VAE isn't a very good baseline to give a good latent space.

**Comment by JingfengYao** [comment]
1. The configuration for KL is provided [here](https://github.com/hustvl/LightningDiT/blob/main/tokenizer/configs/vavae_f16d32.yaml#L15); the latent status is provided [here in README](https://github.com/hustvl/LightningDiT?tab=readme-ov-file#inference-with-pre-trained-models). It is a classic configuration for LDM and is not the main focus of this paper, but its performance on ImageNet has been discussed in recent work [MAR](https://github.com/LTH14/mar/issues/3).
2. The SD3 issue you mentioned has been discussed in our Introduction (page 2).
3. There is a misunderstanding. The number of layers is not the determining factor. The architecture of SD3 differs from that of DiT, and the 22-layer SD3 has a larger number of parameters than DiT-L. According to SD3, it is the model's capability that matters.
4. There is a causal error here. The VAE similar to LDM is the most widely adopted latent diffusion VAE (e.g., SD3, FLUX, SD-XL, DiT), which is why we used it for our experiments.

**Comment by rakkit** [comment]
@JingfengYao Hi, thanks.
Yes, I do agree with you. My point is the baseline is not strong enough to fully show the power of your work, as well as another distillation-based method.  That is also a potential answer to why a higher `D` here did not give better results due to the weak of kl-vae.   Techniquely saying with a stronger tokenizer (better latent landscape), your method can work better

**Comment by rakkit** [comment]
Hey, @JingfengYao, the Vae's checkpoints have not all been released. Is it possible you could share the scale_factor (variance) of  F16D16, F16D32 and F16D64 VAE trained without VF loss?

**Comment by JingfengYao** [comment]
Hi, thanks for your attention. 

We have released more va-vae experimental variants [here](https://huggingface.co/hustvl/va-vae-imagenet256-experimental-variants/tree/main). Hope you like it.  😊
